code/modules/food_and_drinks/kitchen_machinery/griddle.dm

```diff
@@ -63,6 +63,37 @@
 
 
 /obj/machinery/griddle/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/storage/bag/tray))
+		var/datum/component/storage/tray_storage = I.GetComponent(/datum/component/storage)
+		if(!I.contents.len && griddled_objects.len > 0) //if the griddle has things on it and the tray is empty, put what you can into the tray
+			for(var/obj/item/grilled as anything in griddled_objects)
+				if(!griddled_objects.len)
+					return
```

Comment:
because if the griddle is empty i need it to immediately stop, if it doesnt the loop will place the tray onto the griddle, a few of these return ifs are explicitly just to stop that issue, because trays are technically valid objects to be placed onto the griddle, the for loops are constantly attempting to place the tray onto the griddle so i have to account for any instance in which the foor loop continues but cant do what its meant to do, in this case if it cant do what its meant to do, its either empty OR its not empty and it just has an item on it the tray cant grab, ive tried testing with just "else returns" at the end of the foor loop but that still placed the trays onto the griddle..... i honestly dont know WHY since theoretically it shouldnt in that case but it does